portfolio/src/main/webapp/script.js

```diff
@@ -12,15 +12,15 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-
-async function getPhrase() {
-  let response = await fetch('/data');
-  let phrase = await response.text();
-  document.getElementById('dynamic-description').innerText = phrase;
+function getPhrase(){
+    fetch('data').then((response) => response.json()).then((list) => {
+        document.getElementById('dynamic-description').innerText = list;
+    });
 }
```

Comment:
Cool stuff! Wondering why you wanted to switch to a promise style instead of using async like you did above? For a lot of developers, async is typically the more readable pattern but perhaps I'm missing smth here:
https://hackernoon.com/6-reasons-why-javascripts-async-await-blows-promises-away-tutorial-c7ec10518dd9

Reply:
Oh yeah honestly I wanted to try out the different ways of async programming to get a hang of more techniques because I read about promises and callbacks and found them intriguing.

Reply:
But yeah, async/await is definitely quicker to implement

Reply:
Ah okay. Back when I was web developing, promises were the hot thing, and it was a huge improvement to callbacks. The reason being is that callbacks made code super hard to read since it required tons of nested functions.

Now, I think the latest craze is async/await as it makes the code even more readable than promises. I'd say go with async/await in the future but I'm not with the times in javascript. :)

Reply:
Will do :)